lib/course_planner/teachers.ex

```diff
@@ -48,4 +48,14 @@ defmodule CoursePlanner.Teachers do
   defp format_error({:ok, teacher}, _), do: {:ok, teacher}
   defp format_error({:error, changeset}, teacher), do: {:error, teacher, changeset}
 
+  def courses(teacher_id) do
+    Repo.all(from oc in OfferedCourse,
+      left_join: oct in "offered_courses_teachers", on: oct.offered_course_id == oc.id,
+      left_join: u in User, on: u.id == oct.teacher_id,
+      join: t in assoc(oc, :term),
+      preload: [term: t],
+      preload: [:course],
```

Comment:
okay, I see the point of having the preload like that

if you leave like you did ecto can reuse the query to preload the term, so it does one less query to preload it

👍